manifests/mongodb_exporter.pp

```diff
@@ -98,8 +98,40 @@
 
   $options = "${flag_prefix}mongodb.uri=${cnf_uri} ${extra_options}"
 
+  if $install_method == 'url' and (versioncmp($version, '0.7.0') >= 0) {
+    # Not a big fan of copypasting but prometheus::daemon takes for granted
+    # a specific path embedded in the prometheus *_exporter tarball, which
+    # mongodb_exporter lacks.
+    # TODO: patch prometheus::daemon to support custom extract directories
+    $exporter_install_method = 'none'
+    $install_dir = "/opt/${service_name}-${version}.${os}-${arch}"
+    file { $install_dir:
+      ensure => 'directory',
+      owner  => 'root',
+      group  => 0, # 0 instead of root because OS X uses "wheel".
+      mode   => '0555',
+    }
+    -> archive { "/tmp/${service_name}-${version}.${download_extension}":
```

Comment:
We should not be using the arrow syntax between resources like this and should instead use before/require and notify/subscribe metaparameters.

Reply:
Looks like this was no problem two years ago:
* https://github.com/voxpupuli/puppet-prometheus/pull/236/files
* https://github.com/voxpupuli/puppet-archive#puppet-staging-with-zip-file-extraction

Good luck with checking the remaining ~130 modules.